src/workerd/api/queue.h

```diff
@@ -32,30 +32,48 @@ class WorkerQueue: public jsg::Object {
     // contentType determines the serialization format of the message.
     jsg::Optional<kj::String> contentType;
 
-    JSG_STRUCT(contentType);
+    // The number of seconds to delay the delivery of the message being sent.
+    jsg::Optional<uint> delaySecs;
+
+    JSG_STRUCT(contentType, delaySecs);
     JSG_STRUCT_TS_OVERRIDE(QueueSendOptions {
       contentType?: QueueContentType;
+      delaySecs ?: number;
     });
     // NOTE: Any new fields added here should also be added to MessageSendRequest below.
   };
 
+  struct SendBatchOptions {
+    // The number of seconds to delay the delivery of the message being sent.
+    jsg::Optional<uint> delaySecs;
+
+    JSG_STRUCT(delaySecs);
+    JSG_STRUCT_TS_OVERRIDE(QueueSendBatchOptions { delaySecs ?: number; });
+    // NOTE: Any new fields added here should also be added to MessageSendRequest below.
+  };
+
   struct MessageSendRequest {
     jsg::JsRef<jsg::JsValue> body;
 
     // contentType determines the serialization format of the message.
     jsg::Optional<kj::String> contentType;
 
-    JSG_STRUCT(body, contentType);
+    // The number of seconds to delay the delivery of the message being sent.
+    jsg::Optional<uint> delaySecs;
```

Comment:
Do we even want to support setting the delay on each message in a batch individually vs just on an entire batch? I thought the discussion on the PRD was that we were only going to support it on the entire batch. I'll tag you in the thread.